From eb0323c25ea6d3ddab0e85c9f24d29154155ae4f Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <rstoyanov@fedoraproject.org>
Date: Sat, 18 Dec 2021 17:25:17 +0000
Subject: [PATCH 072/249] ci: use unstable release for cross-compile

We added cross-compile tests with testing debian release to be able to
replicate the error reported in #1653, however, installing build
dependencies in this release currently fails with the following error:

libc6-dev:armhf : Breaks: libc6-dev-armhf-cross (< 2.33~) but 2.32-1cross4 is to be installed

This is not something we can fix, therefore using the debian unstable
release (instead of testing) could be more reliable option for our CI.
This would still replicate the problem reported in #1653.

Signed-off-by: Radostin Stoyanov <rstoyanov@fedoraproject.org>
---
 .github/workflows/cross-compile.yml                       | 8 ++++----
 scripts/build/Dockerfile.aarch64-testing-cross.tmpl       | 1 -
 ...ng-cross.hdr => Dockerfile.aarch64-unstable-cross.hdr} | 0
 scripts/build/Dockerfile.aarch64-unstable-cross.tmpl      | 1 +
 scripts/build/Dockerfile.armv7-testing-cross.tmpl         | 1 -
 ...ting-cross.hdr => Dockerfile.armv7-unstable-cross.hdr} | 0
 scripts/build/Dockerfile.armv7-unstable-cross.tmpl        | 1 +
 scripts/build/Dockerfile.mips64el-testing-cross.tmpl      | 1 -
 ...g-cross.hdr => Dockerfile.mips64el-unstable-cross.hdr} | 0
 scripts/build/Dockerfile.mips64el-unstable-cross.tmpl     | 1 +
 scripts/build/Dockerfile.ppc64-testing-cross.tmpl         | 1 -
 ...ting-cross.hdr => Dockerfile.ppc64-unstable-cross.hdr} | 0
 scripts/build/Dockerfile.ppc64-unstable-cross.tmpl        | 1 +
 ....testing-cross.tmpl => Dockerfile.unstable-cross.tmpl} | 2 +-
 scripts/build/Makefile                                    | 4 ++--
 15 files changed, 11 insertions(+), 11 deletions(-)
 delete mode 120000 scripts/build/Dockerfile.aarch64-testing-cross.tmpl
 rename scripts/build/{Dockerfile.aarch64-testing-cross.hdr => Dockerfile.aarch64-unstable-cross.hdr} (100%)
 create mode 120000 scripts/build/Dockerfile.aarch64-unstable-cross.tmpl
 delete mode 120000 scripts/build/Dockerfile.armv7-testing-cross.tmpl
 rename scripts/build/{Dockerfile.armv7-testing-cross.hdr => Dockerfile.armv7-unstable-cross.hdr} (100%)
 create mode 120000 scripts/build/Dockerfile.armv7-unstable-cross.tmpl
 delete mode 120000 scripts/build/Dockerfile.mips64el-testing-cross.tmpl
 rename scripts/build/{Dockerfile.mips64el-testing-cross.hdr => Dockerfile.mips64el-unstable-cross.hdr} (100%)
 create mode 120000 scripts/build/Dockerfile.mips64el-unstable-cross.tmpl
 delete mode 120000 scripts/build/Dockerfile.ppc64-testing-cross.tmpl
 rename scripts/build/{Dockerfile.ppc64-testing-cross.hdr => Dockerfile.ppc64-unstable-cross.hdr} (100%)
 create mode 120000 scripts/build/Dockerfile.ppc64-unstable-cross.tmpl
 rename scripts/build/{Dockerfile.testing-cross.tmpl => Dockerfile.unstable-cross.tmpl} (93%)

diff --git a/.github/workflows/cross-compile.yml b/.github/workflows/cross-compile.yml
index eec6eb577..461a6e618 100644
--- a/.github/workflows/cross-compile.yml
+++ b/.github/workflows/cross-compile.yml
@@ -10,13 +10,13 @@ jobs:
       matrix:
         target: [
           armv7-stable-cross,
-          armv7-testing-cross,
+          armv7-unstable-cross,
           aarch64-stable-cross,
-          aarch64-testing-cross,
+          aarch64-unstable-cross,
           ppc64-stable-cross,
-          ppc64-testing-cross,
+          ppc64-unstable-cross,
           mips64el-stable-cross,
-          mips64el-testing-cross
+          mips64el-unstable-cross
         ]
 
     steps:
diff --git a/scripts/build/Dockerfile.aarch64-testing-cross.tmpl b/scripts/build/Dockerfile.aarch64-testing-cross.tmpl
deleted file mode 120000
index c1dd38b21..000000000
--- a/scripts/build/Dockerfile.aarch64-testing-cross.tmpl
+++ /dev/null
@@ -1 +0,0 @@
-Dockerfile.testing-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.aarch64-testing-cross.hdr b/scripts/build/Dockerfile.aarch64-unstable-cross.hdr
similarity index 100%
rename from scripts/build/Dockerfile.aarch64-testing-cross.hdr
rename to scripts/build/Dockerfile.aarch64-unstable-cross.hdr
diff --git a/scripts/build/Dockerfile.aarch64-unstable-cross.tmpl b/scripts/build/Dockerfile.aarch64-unstable-cross.tmpl
new file mode 120000
index 000000000..955ae1fd4
--- /dev/null
+++ b/scripts/build/Dockerfile.aarch64-unstable-cross.tmpl
@@ -0,0 +1 @@
+Dockerfile.unstable-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.armv7-testing-cross.tmpl b/scripts/build/Dockerfile.armv7-testing-cross.tmpl
deleted file mode 120000
index c1dd38b21..000000000
--- a/scripts/build/Dockerfile.armv7-testing-cross.tmpl
+++ /dev/null
@@ -1 +0,0 @@
-Dockerfile.testing-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.armv7-testing-cross.hdr b/scripts/build/Dockerfile.armv7-unstable-cross.hdr
similarity index 100%
rename from scripts/build/Dockerfile.armv7-testing-cross.hdr
rename to scripts/build/Dockerfile.armv7-unstable-cross.hdr
diff --git a/scripts/build/Dockerfile.armv7-unstable-cross.tmpl b/scripts/build/Dockerfile.armv7-unstable-cross.tmpl
new file mode 120000
index 000000000..955ae1fd4
--- /dev/null
+++ b/scripts/build/Dockerfile.armv7-unstable-cross.tmpl
@@ -0,0 +1 @@
+Dockerfile.unstable-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.mips64el-testing-cross.tmpl b/scripts/build/Dockerfile.mips64el-testing-cross.tmpl
deleted file mode 120000
index c1dd38b21..000000000
--- a/scripts/build/Dockerfile.mips64el-testing-cross.tmpl
+++ /dev/null
@@ -1 +0,0 @@
-Dockerfile.testing-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.mips64el-testing-cross.hdr b/scripts/build/Dockerfile.mips64el-unstable-cross.hdr
similarity index 100%
rename from scripts/build/Dockerfile.mips64el-testing-cross.hdr
rename to scripts/build/Dockerfile.mips64el-unstable-cross.hdr
diff --git a/scripts/build/Dockerfile.mips64el-unstable-cross.tmpl b/scripts/build/Dockerfile.mips64el-unstable-cross.tmpl
new file mode 120000
index 000000000..955ae1fd4
--- /dev/null
+++ b/scripts/build/Dockerfile.mips64el-unstable-cross.tmpl
@@ -0,0 +1 @@
+Dockerfile.unstable-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.ppc64-testing-cross.tmpl b/scripts/build/Dockerfile.ppc64-testing-cross.tmpl
deleted file mode 120000
index c1dd38b21..000000000
--- a/scripts/build/Dockerfile.ppc64-testing-cross.tmpl
+++ /dev/null
@@ -1 +0,0 @@
-Dockerfile.testing-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.ppc64-testing-cross.hdr b/scripts/build/Dockerfile.ppc64-unstable-cross.hdr
similarity index 100%
rename from scripts/build/Dockerfile.ppc64-testing-cross.hdr
rename to scripts/build/Dockerfile.ppc64-unstable-cross.hdr
diff --git a/scripts/build/Dockerfile.ppc64-unstable-cross.tmpl b/scripts/build/Dockerfile.ppc64-unstable-cross.tmpl
new file mode 120000
index 000000000..955ae1fd4
--- /dev/null
+++ b/scripts/build/Dockerfile.ppc64-unstable-cross.tmpl
@@ -0,0 +1 @@
+Dockerfile.unstable-cross.tmpl
\ No newline at end of file
diff --git a/scripts/build/Dockerfile.testing-cross.tmpl b/scripts/build/Dockerfile.unstable-cross.tmpl
similarity index 93%
rename from scripts/build/Dockerfile.testing-cross.tmpl
rename to scripts/build/Dockerfile.unstable-cross.tmpl
index 1d5565310..dacfd96ef 100644
--- a/scripts/build/Dockerfile.testing-cross.tmpl
+++ b/scripts/build/Dockerfile.unstable-cross.tmpl
@@ -1,7 +1,7 @@
 COPY scripts/ci/apt-install /bin/apt-install
 
 # Add the cross compiler sources
-RUN echo "deb http://deb.debian.org/debian/ testing main" >> /etc/apt/sources.list && \
+RUN echo "deb http://deb.debian.org/debian/ unstable main" >> /etc/apt/sources.list && \
   dpkg --add-architecture ${DEBIAN_ARCH}
 
 RUN apt-install \
diff --git a/scripts/build/Makefile b/scripts/build/Makefile
index b24fc80ea..2c006ad87 100644
--- a/scripts/build/Makefile
+++ b/scripts/build/Makefile
@@ -1,7 +1,7 @@
 ARCHES := x86_64 fedora-asan fedora-rawhide centos7 armv7hf centos8
 STABLE_CROSS_ARCHES := armv7-stable-cross aarch64-stable-cross ppc64-stable-cross mips64el-stable-cross
-TESTING_CROSS_ARCHES := armv7-testing-cross aarch64-testing-cross ppc64-testing-cross mips64el-testing-cross
-NON_CLANG := $(TESTING_CROSS_ARCHES) $(STABLE_CROSS_ARCHES)
+UNSTABLE_CROSS_ARCHES := armv7-unstable-cross aarch64-unstable-cross ppc64-unstable-cross mips64el-unstable-cross
+NON_CLANG := $(UNSTABLE_CROSS_ARCHES) $(STABLE_CROSS_ARCHES)
 CREATE_DOCKERFILES := $(ARCHES) $(NON_CLANG)
 TARGETS := $(ARCHES) alpine archlinux
 TARGETS_CLANG := $(addsuffix $(TARGETS),-clang)
-- 
2.35.1

